### PR TITLE
Vsuite

### PR DESCRIPTION
**Pull request by radare**

### CODE REVIEW

File: test/new/r2r-v/r2r.v [hunk 1]
```diff
@@ -496,7 +567,19 @@ fn (r2r mut R2R)load_tests() {
 	dirs := os.ls(db_path) or { panic(err) }
 	for dir in dirs {
 		if dir == 'archos' {
-			println('TODO: archos tests')
+			$if x64 {
```
Comment (ret2libc) [inline test/new/r2r-v/r2r.v]: isn't there a if/elseif in V?
In python, this would look much better with something like:
```
if linux:
  ....
elif macos:
  ....
elif openbsd:
  ....
else:
  print('not supported')
```

Comment (radare) [reply test/new/r2r-v/r2r.v]: not yet. im filling issues and contributing to the compiler, and this is one of the limitations i found

File: test/new/db/archos/linux-x64/dbg_bps [hunk 2]
```diff
@@ -1,7 +1,7 @@
 NAME='dbg.dc+ds.swstep'
 FILE=../bins/elf/analysis/elf-nx
 ARGS=-d
-CMDS=<<EXPECT
+CMDS=<<EOF
```
Comment (ret2libc) [inline test/new/db/archos/linux-x64/dbg_bps]: why this change? wasn't the whole purpose of using <<EXPECT to avoid having to introduce EOF?

File: test/new/r2r-v/r2r.v (hunk 1 above)
Comment (radare) [reply test/new/r2r-v/r2r.v]: this $if thing is a compile time conditional, so the final binary will only contian this code if arch/os matches

File: test/new/db/archos/linux-x64/dbg_bps (hunk 2 above)
Comment (radare) [reply test/new/db/archos/linux-x64/dbg_bps]: the <<EXPECT thing is was just for backward compat for lazines, i want the parser to be simple and be able to move K=V easily upp and down without breaking because the next statement is not a keyword. this was kind of a hack and we must kill it, the asm suite also have a bunch of old syntaxes that im trying to avoid. the v suite is throwing warnings at them

File: test/new/r2r-v/r2r.v (hunk 1 above)
Comment (radare) [reply test/new/r2r-v/r2r.v]: follow it here: https://github.com/vlang/v/issues/3037

Comment (ret2libc) [reply test/new/r2r-v/r2r.v]: ok
